x/group/server/invariants.go

```diff
@@ -149,23 +168,185 @@ func groupTotalWeightInvariant(ctx sdk.Context, groupTable orm.Table, groupMembe
 			}
 			curMemWeight, err := regenMath.ParseNonNegativeDecimal(groupMember.GetMember().GetWeight())
 			if err != nil {
-				return msg, broken, err
+				msg += fmt.Sprintf("error while parsing non-nengative decimal for group member %s\n%v\n", groupMember.Member.Address, err)
+				return msg, broken
 			}
 			err = regenMath.Add(membersWeight, membersWeight, curMemWeight)
 			if err != nil {
-				return msg, broken, err
+				msg += fmt.Sprintf("decimal addition error while adding group member voting weight to total voting weight\n%v\n", err)
+				return msg, broken
 			}
 		}
 		groupWeight, err := regenMath.ParseNonNegativeDecimal(groupInfo.GetTotalWeight())
 		if err != nil {
-			return msg, broken, err
+			msg += fmt.Sprintf("error while parsing non-nengative decimal for group with ID %d\n%v\n", groupInfo.GroupId, err)
+			return msg, broken
 		}
 
 		if groupWeight.Cmp(membersWeight) != 0 {
 			broken = true
-			msg += "group's TotalWeight must be equal to the sum of its members' weights\n"
+			msg += fmt.Sprintf("group's TotalWeight must be equal to the sum of its members' weights\ngroup weight: %s\nSum of group members weights: %s\n", groupWeight.String(), membersWeight.String())
+			break
+		}
+	}
+	return msg, broken
+}
+
+func tallyVotesSumInvariant(ctx sdk.Context, groupTable orm.Table, proposalTable orm.AutoUInt64Table, groupMemberTable orm.PrimaryKeyTable, voteByProposalIndex orm.UInt64Index, groupAccountTable orm.PrimaryKeyTable) (string, bool) {
+	var msg string
+	var broken bool
+
+	var groupInfo group.GroupInfo
+	var proposal group.Proposal
+	var groupAcc group.GroupAccountInfo
+	var groupMem group.GroupMember
+	var vote group.Vote
+
+	proposalIt, err := proposalTable.PrefixScan(ctx, 1, math.MaxUint64)
+	if err != nil {
+		msg += fmt.Sprintf("PrefixScan failure on proposal table at block height %d\n%v\n", ctx.BlockHeight(), err)
```

Comment:
I don't think it's necessary to specify the block height in the msg for this invariant